compare-evidence-strings/compare.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+# A script to compare evidence strings. Please see README.md for details on using it.
+
+# A function to sort key in evidence strings. This makes them more readable and helps comparison through word diff.
+# Also removes the "validated_aginst_schema_version" field entirely, because it frequently changes between the
+# versions, and this change is not important.
+# The two arguments are input and output JSON files.
+sort_keys () {
+  ./jq -S "." --tab <"$1" \
+    | tr -d '\t\n' \
+    | sed -e 's|}{|}~{|g' \
+    | tr '~' '\n' \
+    | sed -e 's|,"validated_against_schema_version": "[0-9.]*"||g' \
+  > "$2"
+}
+
+# A function to extract some fields from the evidence strings. The fields being extracted are:
+# * ClinVar RCV accession
+# * Variant ID (rsID or, if absent, RCV accession)
+# * Ensembl gene ID
+# * Functional consequence SO code
+extract_fields () {
+  ./jq '
+    .unique_association_fields.clinvarAccession + "|" +
+    .unique_association_fields.variant_id + "|" +
+    .unique_association_fields.gene + "|" +
+    .evidence.gene2variant.functional_consequence
+  ' < "$1" \
+  | tr -d '"' | tr '|' '\t' \
+  | sed -e 's|http://purl.obolibrary.org/obo/||g' \
+        -e 's|http://identifiers.org/clinvar.record/||g' \
+        -e 's|http://identifiers.org/dbsnp/||g' \
+        -e 's|http://www.ncbi.nlm.nih.gov/clinvar/||g' \
+  > "$2"
+}
+
+echo "Set up environment & parse parameters"
+export -f sort_keys extract_fields
+# To ensure that the sort results are consistent, set the sort order locale directly
+export LC_COLLATE=C
+# The realpath is required to make the paths work after the working directory change
+OLD_EVIDENCE_STRINGS=$(realpath "$1")
+NEW_EVIDENCE_STRINGS=$(realpath "$2")
+mkdir comparison && cd comparison || exit 1
+
+echo "Install JQ — a command line JSON processor"
+wget -q -O jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
+chmod a+x jq
+
+echo "Install aha — HTML report generator"
+wget -q https://github.com/theZiz/aha/archive/0.5.zip
+unzip -q 0.5.zip && cd aha-0.5 && make && mv aha ../ && cd .. && rm -rf aha-0.5 0.5.zip
```

Comment:
Judgement call: This [library](https://github.com/rtfpessoa/diff2html-cli) seems to be a bit more mature....

Reply:
Agreed, that's a cool library! I added it to the list of future improvements (https://github.com/EBIvariation/eva-cttv-pipeline/pull/95/commits/930bc29115a64d7980d8ec53c0ea92be997d18cd) and will try it out the next time I'll have to compare evidence strings